flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -133,66 +123,22 @@ public boolean isBounded() {
                 getAllPartitions(
                         jobConf,
                         hiveVersion,
                         tablePath,
-                        catalogTable,
-                        hiveShim,
+                        catalogTable.getPartitionKeys(),
                         remainingPartitions);
-        Configuration configuration = Configuration.fromMap(catalogTable.getOptions());
 
-        HiveSource.HiveSourceBuilder sourceBuilder =
-                new HiveSource.HiveSourceBuilder(
-                        jobConf,
-                        tablePath,
-                        catalogTable,
-                        allHivePartitions,
-                        limit,
-                        hiveVersion,
-                        flinkConf.get(HiveOptions.TABLE_EXEC_HIVE_FALLBACK_MAPRED_READER),
-                        (RowType) getProducedDataType().getLogicalType());
-        if (isStreamingSource()) {
-            if (catalogTable.getPartitionKeys().isEmpty()) {
-                PartitionOrder partitionOrder = configuration.get(STREAMING_SOURCE_PARTITION_ORDER);
-                if (partitionOrder != PartitionOrder.CREATE_TIME) {
-                    throw new UnsupportedOperationException(
-                            "Only '"
-                                    + PartitionOrder.CREATE_TIME
-                                    + "' is supported for non partition table.");
-                }
-            }
-
-            Duration monitorInterval =
-                    configuration.get(STREAMING_SOURCE_MONITOR_INTERVAL) == null
-                            ? DEFAULT_SCAN_MONITOR_INTERVAL
-                            : configuration.get(STREAMING_SOURCE_MONITOR_INTERVAL);
-            sourceBuilder.monitorContinuously(monitorInterval);
-
-            if (!catalogTable.getPartitionKeys().isEmpty()) {
-                sourceBuilder.setFetcher(new HiveContinuousPartitionFetcher());
-                final String defaultPartitionName =
-                        jobConf.get(
-                                HiveConf.ConfVars.DEFAULTPARTITIONNAME.varname,
-                                HiveConf.ConfVars.DEFAULTPARTITIONNAME.defaultStrVal);
-                HiveContinuousPartitionFetcherContext<?> fetcherContext =
-                        new HiveContinuousPartitionFetcherContext(
-                                tablePath,
-                                hiveShim,
-                                new JobConfWrapper(jobConf),
-                                catalogTable.getPartitionKeys(),
-                                getTableSchema().getFieldDataTypes(),
-                                getTableSchema().getFieldNames(),
-                                configuration,
-                                defaultPartitionName);
-                sourceBuilder.setFetcherContext(fetcherContext);
-            }
-        }
+        sourceBuilder
+                .setPartitions(allHivePartitions)
```

Comment:
Is it needed to set [artitions in here?  In streaming mode, I think it's no need to set parition for it will try to fetch paritions instead of use the  partitions set; And in batch mode, if not set partitions, by default, HiveSourceBuilder will get all paritions and set the all partitions.

Reply:
You're right that we don't need to call `setPartitions` in streaming mode. But we still need to call it in batch mode. The `allHivePartitions` may not really be **all** partitions, because a partition pruning may have been applied.

Reply:
Can we only call setPartitions in batch mode? So that in streaming mode, we can avoid calling hive metastore to get allHivePartitions.

Reply:
Yes we should. I'll update the PR.